server/game/cards/09.3-TSs/GhostlyCommunion.js

```diff
@@ -0,0 +1,45 @@
+const GameActions = require('../../GameActions/index.js');
+const SpellCard = require('../../spellcard.js');
+
+class GhostlyCommunion extends SpellCard {
+    setupCardAbilities(ability) {
+        this.spellAction({
+            title: 'Noon: Move Dude',
+            playType: 'noon',
+            cost: ability.costs.bootSelf(),
+            difficulty: 5,
+            target: {
+                activePromptTitle: 'Choose a location to move your dude to',
+                waitingPromptTitle: 'Waiting for opponent to choose a location',
+                cardCondition: card => {card.location === 'play area' && (card.hasKeyword('holy ground') || card.adjacentLocations().hasKeyword('holy ground'))},
```

Comment:
You can have a `cardCondition` as a function or object, but you seems to be mixing both.
For target, I usually use object, you can check some other cards:
``` 
cardCondition: { location: 'play area', condition: card => card.hasKeyword('holy ground') || card.adjacentLocations().some(location => location.locationCard.hasKeyword('holy ground') }
```
You see the other problem .. you were treating the `card.adjacentLocations()` result as a card, but it is array.